### DISCUSSION TITLE
3DConnexion Space Mouse, ... in Wine

### DISCUSSION THREAD

**Discussion by krokodyl1220** [cryinkfly/Autodesk-Fusion-360-for-Linux#227] [Q&A] [answered]
Hi,

for a bit of time I am looking for a solution for running Wine apps together with the space mouse (https://3dconnexion.com/pl/spacemouse/). I found some ancient info stating that it is difficult to get USB devices to cooperate. This is the latest thread on Wine forum that I could find (https://forum.winehq.org/viewtopic.php?t=29217), but apart from that - nothing.

What gives me some hope: https://wiki.winehq.org/Hid If using libspnav (https://github.com/FreeSpacenav/libspnav) on Linux I __think__ it should be possible to redirect the input from Linux to Wine. But I have never managed to check it further.

Does anybody know if using 3dmouse in Wine is possible? Or is it a lost cause caused by how Wine works?

**Comment by Eric8tF** [comment]
Hi,

have you had any luck with your device?

Eric

**Comment by krokodyl1220** [comment]
Unfortunately, no.

From what I read previously on the topic it seems like any device that requires the use of custom driver is out of question for use with wine. So my solution is to use VFIO and run a full-blown Windows VM, passing through the USB device and a graphics card. Looking-glass is a great tool for this use case, since I don't have to switch monitor inputs.

For me this work great and I am getting fantastic performance out of the software. If you would like some help with evaluating if this solution is for you please let me know.

**Comment by cryinkfly** [comment, marked as answer]
The 3DConexion Spacemouse works!
See short video: https://www.youtube.com/shorts/CC_w9rxLQZs

Here is also the help from a user from my community and from another project that I will include in my installer.

-> https://github.com/DD1984/SpaceMouse_Fusion360_Wine

**Reply by krokodyl1220** [reply]
Wow, that is so awesome! Thank you for the information, this seems like it could work for other apps also (like Altium Designer).
